mk/config.mk

```diff
@@ -13,11 +13,10 @@ CC  ?= gcc
 CPP ?= cpp
 AR  ?= ar
 # NOTE: gcc-ar is a wrapper around ar that ensures proper integration with GCC plugins,
-# 		such as lto. Using gcc-ar is preferred when creating or linking static libraries
-# 		if the binary is compiled with -flto.
-# 		However, this doesn't apply to darwin as it is using clang instead, and there's no
-# 		gcc-ar wrapper as well.
-CC_AR ?= $(if $(findstring Darwin,$(shell uname -s)),ar,gcc-ar)
+# 	such as lto. Using gcc-ar is preferred when creating or linking static libraries
+# 	if the binary is compiled with -flto. However, it is not universally present, so
+#       only use it if available.
+CC_AR ?= $(if $(shell which gcc-ar),gcc-ar,ar)
```

Comment:
Doesn't this break if you are using Darwin with gcc-ar installed, but you compile with clang?
I'm not very sure about this.

Reply:
I tried to fix this by looking at `cc --version` to see if it's clang or not, and if it's gcc, check for gcc-ar.